src/melee/it/items/ittomato.c

```diff
@@ -0,0 +1,186 @@
+#include "ittomato.h"
+
+#include "baselib/jobj.h"
+#include "it/inlines.h"
+#include "it/it_266F.h"
+#include "it/it_26B1.h"
+#include "it/item.h"
+
+#include <placeholder.h>
+
+extern M2C_UNK gm_80473A18;
+
+/* 2841B4 */ static HSD_GObj* it_802841B4(s32 arg0, Vec3* arg1, s32 arg2);
+/* 28428C */ static void it_8028428C(Item_GObj* gobj);
+/* 284358 */ static void it_80284358(Item_GObj* gobj);
+/* 284380 */ static bool it_80284380(HSD_GObj* gobj);
+/* 284388 */ static void it_80284388(Item_GObj* gobj);
+/* 2843B8 */ static bool it_802843B8(Item_GObj* gobj);
+/* 2843E4 */ static void it_802843E4(HSD_GObj* gobj);
+/* 284420 */ static bool it_80284420(HSD_GObj* gobj);
+/* 284428 */ static void it_80284428(Item_GObj* gobj);
+/* 28442C */ static bool it_8028442C(Item_GObj* gobj);
+/* 284458 */ static void it_80284458(HSD_GObj* gobj);
+/* 284480 */ static bool it_80284480(Item_GObj* gobj);
+/* 284488 */ static void it_80284488(Item_GObj* gobj);
+/* 2844B8 */ static bool it_802844B8(Item_GObj* gobj);
+/* 284540 */ static bool it_80284540(Item_GObj* gobj);
+/* 284548 */ static void it_80284548(Item_GObj* gobj);
+/* 2845E4 */ static bool it_802845E4(Item_GObj* gobj);
+/* 2845EC */ static void it_802845EC(Item_GObj* gobj);
+/* 2845F0 */ static bool it_802845F0(Item_GObj* gobj);
+
+HSD_GObj* it_802841B4(s32 arg0, Vec3* arg1, s32 arg2)
+{
+    NOT_IMPLEMENTED;
+}
+
+void it_8028428C(Item_GObj* gobj)
+{
+    Item* ip;
+
+    ip = gobj->user_data;
```

Comment:
Declarations can usually be combined without affecting codegen, and stylistically it's preferred.
```suggestion
    Item* ip = gobj->user_data;
```

If it doesn't break the match, this is also preferred:
```suggestion
    Item* ip = GET_ITEM(gobj);
```